league_rpc_linux/processes/lcu_thread.py

```diff
@@ -1,355 +1,370 @@
 
-
-def update_rpc_and_reset_flag():
-    global scheduledUpdate
-    update_rpc()
-    scheduledUpdate = False
+    def update_rpc_and_reset_flag(self):
+        self.lcu.update_rpc()
+        self.update_already_scheduled = False
 
 
 ## WS Events ##
 
 
-@connector.ready
-async def connect(connection):
-    print(f"{Colors.green}LCU API is ready.{Colors.reset}")
-
-    await gather_based_data(connection)
-    delayUpdate()
-
-
-@connector.close
-async def disconnect(connection):
-    print("LCU API is closed.")
-
-
-@connector.ws.register("/lol-summoner/v1/current-summoner", event_types=("UPDATE",))
-async def summoner_updated(connection, event):
-    print("Summoner has been updated.")
-    data["summoner_name"] = event.data["displayName"]
-    data["level"] = event.data["summonerLevel"]
-    data["summoner_id"] = event.data["summonerId"]
-    data["summoner_icon"] = event.data["profileIconId"]
-
-    delayUpdate()
-
-
-@connector.ws.register("/lol-chat/v1/me", event_types=("UPDATE",))
-async def chat_updated(connection, event):
-    print("Chat has been updated.")
-    if event.data["availability"] == "chat":
-        data["availability"] = "Idle"
-    if event.data["availability"] == "away":
-        data["availability"] = "Away"
-
-    delayUpdate()
-
-
-@connector.ws.register("/lol-gameflow/v1/gameflow-phase", event_types=("UPDATE",))
-async def gameflow_phase_updated(connection, event):
-    print("Gameflow Phase has been updated.")
-    data["gameflowPhase"] = event.data  # returns plain string of the phase
+class LcuThread:
+    self_obj: None
+
+    def __init__(self, rpc_from_main):
+        self.rpc = rpc_from_main
+        self.connector = lcu_driver.Connector()
+        LcuThread.connector = self.connector
+
+        self.update_scheduler = UpdateScheduler(self)
+        self.data = {
+            "champion_name": None,
+            "skin_name": None,
+            "skin_id": None,
+            "level": None,
+            "gamemode": None,
+            "is_custom": False,
+            "is_practice": False,
+            "in_lobby": False,
+            "in_queue": False,
+            "in_champ_select": False,
+            "max_players": None,
+            "players": None,
+            "summoner_name": None,
+            "summoner_id": None,
+            "summoner_icon": None,
+            "summoner_rank": None,
+            "queue": None,
+            "queue_id": -1,
+            "map_id": None,  # 11, 12, 21, 22, 30
+            "is_ranked_queue": False,  # True, False
+            "lobby_id": None,
+            "availability": "Idle",  # Idle, Away
+            "gamflow_phase": None,  # Lobby, Matchmaking, ReadyCheck, ChampSelect, InProgress, WaitingForStats, EndOfGame
+        }
+
+        self.register_ws_events(self.connector)
+        print("Starting LCU API.")
+
+    @staticmethod
+    def start_connector(rpc_from_main):
+        lcu_thread = LcuThread(rpc_from_main)
+        lcu_thread.connector.start()
```

Comment:
from the def __init__ you have already called Connector() and stored it in an instance variable ""self.connector"

You could continue to use that instance variable by using "self.connector.start()" instead of creating an entire new LcuThread i believe

Reply:
changed connector.start inside the init